.github/workflows/dependencies-head.yml

```diff
@@ -84,6 +84,31 @@ jobs:
       run: |
         python -m pytest -r sx --ignore tests/benchmarks/ --ignore tests/contrib --ignore tests/test_notebooks.py
 
+  uproot4:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install --ignore-installed --upgrade -q --no-cache-dir -e .[test]
+        python -m pip uninstall --yes uproot
+        python -m pip install --upgrade --no-cache-dir git+git://github.com/scikit-hep/uproot4.git
```

Comment:
Is there a reason you're not getting this from pip?

```bash
python -m pip install --upgrade --no-cache-dir "uproot>=4.0.0rc1"
```

The main branch of the scikit-hep/uproot4 repo is _supposed_ to always be working, but actual releases have a stronger guarantee and unchanging version numbers (to debug a failure with `pip list` in the next line). The version number of the main branch is either the latest release (because I just released it and haven't updated the version yet) or what I expect the next release will be, but I could change my mind about what level of semantic versioning to update. Also, the code in the main branch changes with each PR merge, but the version number does not. Two runs of your CI might get different Uproot code but report the same version number in `pip list` on the next line.

The `>=4.0.0rc1` in the pip-install ensures that you get the latest release candidate, if that's what you're trying to ensure. (Although I think I'm only doing release candidates on Awkward Array, not Uproot.)

Reply:
@jpivarski Yes, this is intentional. This is a nightly test that is supposed to grab whatever is at the HEAD of GitHub (hence the name of the workflow) to give us a **very** early heads up on API breaking behavior. We'd far prefer to be able to check out what is going on and prepare a PR months in advance to deal with breaking behavior then to get busy and miss a release candidate.

This also allows us to check that changes people are making for us do what we wanted (such as when we were testing iminuit changes Hans was making for us).